### DISCUSSION TITLE
Help CDi

### DISCUSSION THREAD

**Discussion by pkerga** [darthcloud/BlueRetro#597] [General]
hi Jacques, 
tonight I'm going to start mounting an adapter for my brand new CDI.  2 questions: 
- if I don't wire a 2nd plug for the back connector, can I use the P2 RXD and P2 RTS of the front connector for a 2nd player?  
- If I'm not wiring 2nd port plug do I have to connect the io21, the ground AND the pin6 of the 2nd 74AHCT125N? 
Thanks

**Comment by pkerga** [comment]
And an other question :
Can Blueretro emulate a controler like this one ? What about the 3 buttons ?
![Screenshot_20230203-202119](https://user-images.githubusercontent.com/30766052/216689424-9e192abf-0f1e-48f5-8fda-6fa93ad9bab6.png)

**Comment by darthcloud** [comment]
CD-i 2nd player pins are always at the back connector if one exist.
Two player on one port is only for system with only one front port.

You always need the ground not matter what you do.

You can skip anything related to the the 2nd port.

Cd-i protocol only expose a joystick and 2 button, so this controller 3rd button is only pressing button 1 + 2 simultaneously 

Load the "CD-i Gamepad" preset with https://blueretro.io/presets.html and everything will work ok.

Btw BlueRetro need to be powered on via external power (like USB) before powering the CD-i.

**Reply by pkerga** [reply]
Thanks !
Here is my first tests. The pointer move alone... the movement is very fine.
Do you think i've donne something wrong ? My Blueretro setting to refine?

https://youtu.be/kUClaPbvAa4

**Comment by pkerga** [comment]
I've made more tests today. 
The strange pointer behaviour is present with wiiU controler and Xbox One. Not with the Switch Pro. 
Not easy to see on the video with my CRT...

https://youtu.be/LBXihk3GBWc
